src/oidc/error.ts

```diff
@@ -0,0 +1,14 @@
+export class OIDCError extends Error {
+    constructor(message: string) {
+        super(message);
+        this.name = 'OIDCError';
+    }
+}
+
+export const OIDCAuthenticationError = {
+    FailedToFetchOIDCConfiguration: new OIDCError('Failed to fetch OIDC configuration'),
```

Comment:
Was thinking to have an error code instead of message to differentiate error better